### PR TITLE
Standalone subscription

### PR DESCRIPTION
**Pull request by fil512**
This pull request moves code out of hapi-fhir-jpaserver-base and into a number of sub-projects:

hapi-fhir-jpaserver-model - entity classes required by searchparam
hapi-fhir-jpaserver-searchparam - searchparam services such as registry and extractor
hapi-fhir-jpaserver-subscription - subscription services that do not require a database e.g. inmemory

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDaoTest.java [hunk 1]
```diff
@@ -58,9 +62,9 @@ public void testTranslateMatchUrl() {
 	
 	@Test
 	public void testNormalizeString() {
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDaoTest.java]: I guess this method should probably be in a new unit test class specifically for StringNormalizer (and be careful that the unicode chars don't get stripped accidentally if you do this)

File: hapi-fhir-jpaserver-subscription/pom.xml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+					<artifactId>maven-site-plugin</artifactId>
+					<configuration>
+						<skipDeploy>true</skipDeploy>
+					</configuration>
+				</plugin>
+			</plugins>
+		</pluginManagement>
+		<plugins>
+			<plugin>
+				<groupId>org.jacoco</groupId>
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-subscription/pom.xml]: Ah, this reminds me- you probably need to update `hapi-fhir-jacoco/pom.xml` to account for the new projects/modules you have added

File: example-projects/hapi-fhir-jpaserver-example-postgres/src/main/java/ca/uhn/fhir/jpa/demo/FhirServerConfig.java [hunk 3]
```diff
@@ -64,12 +64,12 @@ public DataSource dataSource() {
 	}
 
 	@Override
-	@Bean()
+	@Bean
 	public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
```
[the remark this answers is not in the document]
Comment (fil512) [reply example-projects/hapi-fhir-jpaserver-example-postgres/src/main/java/ca/uhn/fhir/jpa/demo/FhirServerConfig.java]: Fixed in 58938926dc673626a2f3fa74ae9047e2b549744d

[the remark this answers is not in the document]
Comment (fil512) [reply example-projects/hapi-fhir-jpaserver-example-postgres/src/main/java/ca/uhn/fhir/jpa/demo/FhirServerConfig.java]: Fixed in 58938926dc673626a2f3fa74ae9047e2b549744d

File: hapi-fhir-cli/hapi-fhir-cli-jpaserver/src/main/java/ca/uhn/fhir/jpa/demo/FhirServerConfig.java [hunk 4]
```diff
@@ -47,12 +47,12 @@ public DaoConfig daoConfig() {
...
 
 	@Override
-	@Bean()
+	@Bean
 	public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
 		LocalContainerEntityManagerFactoryBean retVal = super.entityManagerFactory();
 		retVal.setPersistenceUnitName("HAPI_PU");
 		retVal.setDataSource(myDataSource);
-		retVal.setPackagesToScan("ca.uhn.fhir.jpa.entity");
+		retVal.setPackagesToScan("ca.uhn.fhir.jpa.model.entity", "ca.uhn.fhir.jpa.entity");
```
[the remark this answers is not in the document]
Comment (fil512) [reply hapi-fhir-cli/hapi-fhir-cli-jpaserver/src/main/java/ca/uhn/fhir/jpa/demo/FhirServerConfig.java]: Fixed in 58938926dc673626a2f3fa74ae9047e2b549744d

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDaoTest.java (hunk 1 above)
Comment (fil512) [reply hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDaoTest.java]: Fixed in 5d75b2c636cfe484f7ebc00e5dfada478eec84af

File: hapi-fhir-jpaserver-subscription/pom.xml (hunk 2 above)
Comment (fil512) [reply hapi-fhir-jpaserver-subscription/pom.xml]: Fixed in 3128072decdf645efc780b4cc65e0a931bc06ae6

Comment (fil512) [reply hapi-fhir-jpaserver-subscription/pom.xml]: We'll try again sir... 5caf64bdf49d59db3aa4bbe45198de3e5a6fe1d6

**Comment by fil512** [comment]
Move code out of hapi-fhir-jpaserver-base and into a number of sub-projects:

hapi-fhir-jpaserver-model - entity classes required by searchparam
hapi-fhir-jpaserver-searchparam - searchparam services such as registry and extractor
hapi-fhir-jpaserver-subscription - subscription services that do not require a database e.g. inmemory

The hapi-fhir-jpaserver-subscription depends on the other two and can perform in-memory subscription matching without access to a database.

**Comment by fil512** [comment]
Build failed in an intermittent test failure in hapi-fhir-jpaserver-base.  I confirmed all tests in that module passed in IntelliJ.  I will continue to run builds to confirm that it is possible to get a clean run on this version.

**Pull request merged**